packages/block-library/src/post-template/edit.js

```diff
@@ -155,6 +160,18 @@ export default function PostTemplateEdit( {
 				if ( templateSlug?.startsWith( 'archive-' ) ) {
 					query.postType = templateSlug.replace( 'archive-', '' );
 					postType = query.postType;
+				} else if (
+					templateSlug?.startsWith( 'category-' ) &&
+					!! categories
+				) {
+					const targetSlug = templateSlug.replace( 'category-', '' );
+					const categoryIds = categories
+						?.filter( ( { slug } ) => slug === targetSlug )
+						.map( ( { id } ) => id );
+					query.taxQuery = {
```

Comment:
This is useful, though we need to make it so _any_ taxonomy works by default. I'm thinking on tags but also any 3rd party taxonomies (e.g.: product categories that Woo uses, etc).